### PR TITLE
Nav Goal + FSM

### PR DESCRIPTION
**Pull request by matthew-reynolds**

### CODE REVIEW

File: qpup_vision/launch/zed_ar_track.launch [hunk 1]
```diff
@@ -29,5 +29,7 @@
             <remap from="camera_image"  to="$(arg zed_node_name)/rgb/image_rect_color" />
             <remap from="camera_info"   to="$(arg zed_node_name)/rgb/camera_info" />
         </node>
+
+        <node name="qpup_vision" pkg="person_tracker" type="person_tracker"/>
```
Comment (wmmc88) [inline qpup_vision/launch/zed_ar_track.launch]:
```suggestion
        <node name="person_tracker" pkg="person_tracker" type="person_tracker"/>
```

File: qpup_vision/src/person_tracker.cpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+ros::Subscriber ar_track_sub;
+std_msgs::String state;
+ros::Publisher state_pub;
+move_base_msgs::MoveBaseGoal goal;
+std::unique_ptr<MoveBaseClient> ac;
+
+tf2_ros::Buffer buffer_;
+
+enum class State { FOLLOW, STOP, UNKNOWN };
```
Comment (wmmc88) [inline qpup_vision/src/person_tracker.cpp]:
```suggestion
enum class State { UNKNOWN , FOLLOW, STOP };
```

not that it really matters, but default constructing will zero-init (first index)

File: qpup_vision/src/person_tracker.cpp [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        if (new_state == State::UNKNOWN) {
+          new_state = State::FOLLOW;
+        }
+        new_goal = buffer_.transform(marker.pose, "base_link");
+        break;
+      case 2:
+        new_state = State::STOP;
+        break;
+      default:
+        // N/A
```
Comment (wmmc88) [inline qpup_vision/src/person_tracker.cpp:43]: `ROS_ERROR` or `ROS_WARN`

File: qpup_vision/src/person_tracker.cpp [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        break;
+    }
+  }
+
+  // Offset pose by 2m in the direction of the base_link
```
Comment (wmmc88) [inline qpup_vision/src/person_tracker.cpp:48]: whats this offset for?

File: qpup_vision/src/person_tracker.cpp [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+  // Wait for the action server to come up
+  while (!ac->waitForServer(ros::Duration(5.0))) {
+    ROS_INFO("Waiting for the move_base action server to come up");
+  }
+
+  ar_track_sub = nh.subscribe("/zed/ar_pose_marker", 100, pose_marker_callback);
+  state_pub = nh.advertise<std_msgs::String>("state", 100);
+
+  ros::spin();
```
Comment (wmmc88) [inline qpup_vision/src/person_tracker.cpp:101]: while ros ok, spin once

Comment (wmmc88) [reply qpup_vision/src/person_tracker.cpp:101]: also  fixed rate spin

File: qpup_vision/src/person_tracker.cpp (hunk 4 above)
Comment (matthew-reynolds) [reply qpup_vision/src/person_tracker.cpp:48]: So we walk to the person, not into them

**Comment by matthew-reynolds** [comment]
Untested but eh. Merge then debug later :P

**Review by wmmc88 — APPROVED** [review]

**Pull request merged**